content/blog/2022-03-31-training-and-saving-models-on-a-dedicated-runner.md

```diff
@@ -0,0 +1,276 @@
+[CML getting started guide](https://cml.dev/doc/start/github). Here we create a
+simple `RandomForestClassifier()` based on some generated data. We then use the
+model to make some predictions and plot those predictions in a confusion matrix.
+
+While running the script the model is kept in memory, meaning it is discarded as
+soon as the script finishes. In order to save the model for later, we need to
+dump it as a binary file. We do so with `joblib.dump()`. Later we can read the
+model using `joblib.read()` when we need to.
+
+<admon type="tip">
+You can also use <code>pickle.dump()</code> if you prefer.
+</admon>
+
+The outputs of `train.py` are:
+
+- `metrics.txt`: a file containing metrics on model performance (in this case
+  accuracy)
+- `confusion_matrix.png`: a plot showing the classification results of our model
+- `random_forest.joblib`: the binary output of the trained model
+
+All of these files are saved to the `model` directory.
+
+```python
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import plot_confusion_matrix
+import matplotlib.pyplot as plt
+import json
+import os
+import joblib
+import numpy as np
+
+# Read in data
+X_train = np.genfromtxt("data/train_features.csv")
+y_train = np.genfromtxt("data/train_labels.csv")
+X_test = np.genfromtxt("data/test_features.csv")
+y_test = np.genfromtxt("data/test_labels.csv")
+
+# Fit a model
+depth = 5
+clf = RandomForestClassifier(max_depth=depth)
+clf.fit(X_train,y_train)
+
+# Calculate accuracy
+acc = clf.score(X_test, y_test)
+print(acc)
+
+# Create model folder if it does not yet exist
+if not os.path.exists('model'):
+    os.makedirs('model')
+
+# Write metrics to file
+with open("model/metrics.txt", 'w+') as outfile:
+        outfile.write("Accuracy: " + str(acc) + "\n")
+
+# Plot confusion matrix
+disp = plot_confusion_matrix(clf, X_test, y_test, normalize='true',cmap=plt.cm.Blues)
+plt.savefig('model/confusion_matrix.png')
+
+# Save the model
+joblib.dump(clf, "model/random_forest.joblib")
+```
+
+# Train the model on a daily basis
+
+Now that we have a script to train our model and save it as a file, let’s set up
+our CI/CD to provision a runner and run the script. We define our workflow in
+`cml.yaml` and save it in the `.github/workflows` directory. This way GitHub
+will automatically go through the workflow whenever it is triggered. In this
+case the triggers are a manual run and the daily schedule.
+
+<admon type="info">
+The name of the workflow doesn’t matter, as long as it’s a <code>.yaml</code> and
+located in the <code>.github/workflows</code> directory. You can have multiple workflows
+in there as well. You can learn more in the
+<a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions">documentation</a>
+here.
+</admon>
+
+```yaml
+name: CML
+on: # Here we use two triggers: manually and daily at 08:00
+  workflow_dispatch:
+  schedule:
+    - cron: '0 8 * * *'
```

Comment:
the example repo (https://github.com/iterative/example_model_export_cml) only has `workflow_dispatch` - is the mismatch intentional?

Actually there are other slight differences too. Should probably sync them?